dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/StaticLinkageCheckerTest.java

```diff
@@ -409,6 +412,57 @@ public void testCheckLinkageErrorMissingClassAt_validClassInJar()
     Truth8.assertThat(classSymbolError).isEmpty();
   }
 
+  @Test
+  public void testCheckLinkageErrorMissingClassAt_invalidSuperclass()
+      throws IOException, URISyntaxException {
+    List<Path> paths =
+        ImmutableList.of(
+            absolutePathOfResource("testdata/grpc-google-cloud-firestore-v1beta1-0.28.0.jar"));
+    StaticLinkageChecker staticLinkageChecker =
+        StaticLinkageChecker.create(false, paths, ImmutableSet.copyOf(paths));
+
+    ClassSymbolReference invalidClassReference =
+        ClassSymbolReference.builder()
+            .setSourceClassName("com.google.firestore.v1beta1.FirestoreGrpc") // dummy value
+            .setSubclass(true) // invalid because FirestoreGrpc is a final class
+            .setTargetClassName("com.google.firestore.v1beta1.FirestoreGrpc")
+            .build();
+
+    Optional<StaticLinkageError<ClassSymbolReference>> classSymbolError =
+        staticLinkageChecker.checkLinkageErrorMissingClassAt(invalidClassReference);
+    Truth8.assertThat(classSymbolError).isPresent();
+    Truth.assertThat(classSymbolError.get().getReason())
+        .isEqualTo(Reason.INCOMPATIBLE_CLASS_CHANGE);
+  }
+
+  @Test
+  public void testCheckLinkageErrorMissingClassAt_invalidMethodOverriding()
```

Comment:
This test case catches a real problem [somebody faced in past](https://stackoverflow.com/questions/21059019/cglib-is-causing-a-java-lang-verifyerror-during-query-generation-in-intuit-partn).